crates/flashblocks/p2p/tests/peer_monitoring.rs

```diff
@@ -0,0 +1,467 @@
+use alloy_genesis::Genesis;
+use async_trait::async_trait;
+use ed25519_dalek::SigningKey;
+use eyre::eyre;
+use flashblocks_cli::FlashblocksArgs;
+use flashblocks_node::{FlashblocksNode, FlashblocksNodeBuilder};
+use flashblocks_p2p::events::{NetworkPeersContext, PeerEventsListener};
+use reth_network::events::PeerEvent;
+use reth_network::types::PeerKind;
+use reth_network::{Peers, PeersInfo};
+use reth_network_peers::{NodeRecord, PeerId};
+use reth_node_builder::{Node, NodeBuilder, NodeConfig, NodeHandle};
+use reth_node_core::args::{DiscoveryArgs, NetworkArgs, RpcServerArgs};
+use reth_optimism_chainspec::OpChainSpecBuilder;
+use reth_provider::providers::BlockchainProvider;
+use reth_tasks::{TaskExecutor, TaskManager};
+use std::any::Any;
+use std::net::SocketAddr;
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex,
+};
+use std::time::Duration;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer};
+
+/// Context for a running flashblocks node in tests
+struct NodeContext {
+    pub local_node_record: NodeRecord,
+    _node_exit_future: reth_node_core::exit::NodeExitFuture,
+    _node: Box<dyn Any + Sync + Send>,
+}
+
+async fn setup_flashblocks_node(
```

Comment:
We should either move these tests into the `falshblocks-node` crate, or create a new `flashblocks-test` which we can import into our integration tests